### PR TITLE
Improve interaction error message

### PR DESCRIPTION
**Pull request by LaughLax**
Okay, trying again. This is a "duplicate" of [PR 20](https://github.com/Gaiwecoor/icarus5/pull/20), resubmitted after a reversion so that better review can happen. **Please, nobody approve this PR until these changes have been discussed with and approved by Gaiwecoor.**

This PR aims to make the error messages for interactions more useful by including more information about the interaction. It also includes a bugfix replacing a `message.deferred` check with `message.deferred || message.replied`.

* Are any changes to config.json necessary? **No**
* Are all referenced snowflakes contained in the config/snowflake.json file, rather than in the code directly? **No snowflakes referenced**
* Are any changes to Interaction fingerprints needed? **No**
* Have I used JSdoc style comments (at least) to properly document all of my functions?  **Yes**
* Are there any new dependencies that need to be installed? **No**

### CODE REVIEW

File: utils/utils.js [hunk 1]
```diff
@@ -3,6 +3,70 @@ const Discord = require("discord.js"),
...
+
+    if (options.data[0].type == "SUB_COMMAND") {
+      commandParts.push(options.data[0].name);
+      options = options.data[0].options
+      return {
+        command: commandParts.join(" "),
+        data: options.data
+      };
+    }
+  }
```
Comment (Gaiwecoor) [inline utils/utils.js:47]: I believe `options.data` nests if it's a sub command group or sub command. For example:
```js
options.data = [{
  type: "SUB_COMMAND_GROUP",
  name: "game",
  options: [{
    type: "SUB_COMMAND",
    name: "redeem",
    options: [{
      name: "code",
      type: "STRING",
      value: "SOMECODE"
    }]
  }]
}]
```
If it's nested like I think it is, it'll mess up parsing down around lines 173

**Review by Gaiwecoor — COMMENTED** [review]
Mostly good, one concern.

File: utils/utils.js (hunk 1 above)
Comment (LaughLax) [reply utils/utils.js:47]: See lines 30 and 41 that are intended to address this. Do you think those lines will do the job?

Comment (Gaiwecoor) [reply utils/utils.js:47]: I mean ... if you want to get *technical* about it! I overlooked that.

**Review by Gaiwecoor — APPROVED** [review]

**Pull request merged**